Jimmba/nodejs-aws-cart-api
# Task9: containerization

**Evaluation criteria (70 points for covering all criteria)**

>Dockerfile is prepared, image is building. Image size is minimized to be less than 500 MB:

check the code (Dockerfile)
![image](https://github.com/Jimmba/nodejs-aws-cart-api/assets/32487200/841d6137-357d-4898-897b-b13bc2c689cc)


> Dockerfile is optimized. Files that change more often and commands that depend on them should be included later, files and commands that change less should be at the top.

check the code (Dockerfile)

>Folders are added to .dockerignore, with explanations. At least 2 big directories should be excluded from build context. Elastic Beanstalk application is initialized.

check the code (.dockerignore)
![image](https://github.com/Jimmba/nodejs-aws-cart-api/assets/32487200/ca61e9c2-ee40-40cf-a33a-c894944daa95)


**Additional (optional) tasks**
> 15 - Environment is created and the app is deployed to the AWS cloud. You must provide a link to your GitHub repo with Cart Service API or PR with created Dockerfile and related configurations.

done. Screenshot with BE higher, Card Service repo is [here](https://github.com/Jimmba/nodejs-aws-cart-api)

> 15 - FE application is updated with Cart API endpoint. You must provide a PR with updates in your FE repository and OPTIONALLY link to deployed front-end app which makes proper API calls to your Cart Service.

[FE PR](https://github.com/Jimmba/nodejs-aws-shop-react/pull/4)
[deployed FE APP](https://d8rdhpyuoxkyw.cloudfront.net/)

### HOW TO TEST:

you should add `authorization_token` in local storage. By pressing buttons you can add products to card or change it count
![image](https://github.com/Jimmba/nodejs-aws-cart-api/assets/32487200/62a09a7d-3f03-476c-b7f4-287e69dbc850)

By pressing buttons count updates in database
![image](https://github.com/Jimmba/nodejs-aws-cart-api/assets/32487200/300a1f39-af69-4c98-8cf1-a1bade7ad049)

Open the main page, refresh it, open cart - all data is saved.

NOTE: I added api gateway cause EB uses http while APP uses https, that is forbidden by browsers.
Direct lint to EB is [here](http://jimmba-cart-api-development.eu-west-1.elasticbeanstalk.com). Don't forget about authorization_token
![image](https://github.com/Jimmba/nodejs-aws-cart-api/assets/32487200/342ba67c-7c02-44b8-94c7-7fe8bd413010)

TOTAL 100/100
### Thanks for review and Happy New Year!
